### PR TITLE
Add Style Guide to Docs

### PR DESCRIPTION
**Pull request by mcdappdev**
Moves the style guide from vapor-community/styleguide to inside the docs per the discussion in Discord.

One thing to consider: Should each section be broken down into its own file to make it not a huge document?

Todo - update providers section per discussion from main repo

### CODE REVIEW

**Comment by Mordil** [comment]
This is missing a section in folder structure about “services”

**Comment by mcdappdev** [comment]
Commit 471e9c7 is dependent on Fluent 3.1 supporting a `ConnectionPool` as a `DatabaseConnectable`

File: 3.0/mkdocs.yml [hunk 1]
```diff
@@ -107,6 +107,8 @@ pages:
 - 'WebSocket': 
     - 'Getting Started': 'websocket/getting-started.md'
     - 'Overview': 'websocket/overview.md'
+- 'Extras': 
+    - 'extras/styleguide.md'
```
Comment (tanner0101) [inline 3.0/mkdocs.yml]: This should have a label: `Style Guide`. Also the document should probably be `style-guide.md`.

File: 3.0/docs/extras/styleguide.md [hunk 2]
```diff
@@ -0,0 +1,1063 @@
+# Vapor Style Guide
+
+## Motivation
+
+The Vapor style guide is a perspective on how to write Vapor application code that is clean, readable, and maintainable. It can serve as a jumping off point within your organization (or yourself) for how to write code in a style that aligns with the Vapor ecosystem. We think this guide can help solidify common ideas that occur across most applications and will be a reference for maintainers when starting a new project. This style guide is opinionated, so you should adapt your code in places where you don’t agree.
+
+## Maintainers
```
Comment (tanner0101) [inline 3.0/docs/extras/styleguide.md]: This info should probably go at the bottom.

**Comment by penny-coin** [comment]
Hey @mcdappdev, you just merged a pull request, have a coin! 

You now have 217 coins.

**Pull request merged**